event-app-ui/src/components/Directions/Directions.jsx

```diff
@@ -1,39 +1,123 @@
   const [directionsResponse, setDirectionsResponse] = useState(null);
   const [distance, setDistance] = useState("");
   const [duration, setDuration] = useState("");
   const [center, setCenter] = useState({ lat: 0, lng: 0 });
   const [currentAddress, setCurrentAddress] = useState("");
+  const [weatherInfo, setWeatherInfo] = useState(null);
+  const [weatherNotice, setWeatherNotice] = useState(null);
+  const [destination, setDestination] = useState("");
+  const [startTime, setStartTime] = useState("");
+  const [latestDeparture, setLatestDeparture] = useState(null);
+  const [mode, setMode] = useState("");
+  const { id } = useParams();
+
+  //function that will calculate the difference between the event's start time and the current Date in seconds
+  const secondsBetweenEpochs = (firstEpoch, secondEpoch) => {
+    return (
+      Math.abs(new Date(firstEpoch * 1000) - new Date(secondEpoch * 1000)) /
+      1000
+    );
+  };
+
+  //fetching the location and start time from the The endpoint FIND_EVENT_ENDPOINT_URL
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const result = await axios.get(`${FIND_EVENT_ENDPOINT_URL}/${id}`);
+        setDestination(result.data.location);
+        setStartTime(result.data.starttime);
+
+        //calculates the difference between the start time and the current date
+        //converts it to seconds and rounds it up to get the days between the start time and current date
+        const daysFromNowTillEvent = Math.floor(
+          secondsBetweenEpochs(result.data.starttime, Date.now() / 1000) /
+            SECONDS_IN_A_DAY
+        );
 
-  const destination = address.replace(/%20/g, " ");
-  const { isLoaded } = useJsApiLoader({
-    googleMapsApiKey: API_KEY,
-  });
+        // Forecast API only shows up to 5 days forecast every 3 hours, so we want to check for
+        // what the weather forecast is for 3, 4, or 5 hours before the event start time
+        if (daysFromNowTillEvent <= 5) {
+          const results = await geocodeByAddress(result.data.location);
+
+          //gets the latitude and longitude coordinates
+          const { lat, lng } = await getLatLng(results[0]);
+          const response = await fetch(
+            `${FORCAST_API.base}forecast?lat=${lat}&lon=${lng}&appid=${WEATHER_API.key}`
+          );
+          const data = await response.json();
+
+          //iterate through every data in the data list to get the dt which is the date and time in epoch format
+          //and gets the hours by finding the difference between the start time and end time in seconds
+          for (let info of data.list) {
+            const hoursFromNowTillEvent = Math.floor(
+              secondsBetweenEpochs(info.dt, result.data.starttime) /
+                SECONDS_IN_AN_HOUR
+            );
+
+            if (
+              hoursFromNowTillEvent === 3 ||
+              hoursFromNowTillEvent === 4 ||
+              hoursFromNowTillEvent === 5
```

Comment:
I think you just want hoursFromNowTillEvent <= 3

Reply:
Okay i'll look into that